core/fetcher_test.go

```diff
@@ -14,9 +13,9 @@ func TestBlockFetcher_GetBlock_and_SubscribeNewBlockEvent(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second*15)
 	t.Cleanup(cancel)
 
-	host, port, err := net.SplitHostPort(StartTestNode(t).GRPCClient.Target())
-	require.NoError(t, err)
-	client := newTestClient(t, host, port)
+	cfg := DefaultTestConfig()
+	StartTestNodeWithConfig(t, cfg)
+	client, err := newCometGRPCConn(cfg.TmConfig.RPC.GRPCListenAddress)
```

Comment:
needs to be separated out as `DefaultTestConfig` dynamically allocates ports so we need to keep a reference to it in order to have the correct `TmConfig.RPC.GRPCListenAddress`